dataland-frontend/src/components/resources/chooseFrameworkForDataUpload/MetaInfoPerCompanyAndFramework.vue

```diff
@@ -0,0 +1,130 @@
+      </div>
+
+      <div v-else>
+        <div v-for="(dataMetaInfo, index) in listOfFrameworkData" :key="index">
+          <p
+            :class="[isFrontendViewPageExisting ? ['text-primary', 'cursor-pointer'] : '']"
+            class="font-semibold"
+            @click="redirectToViewPageIfEnabledInFrontend(dataMetaInfo)"
+          >
+            {{ dynamicDatasetTitle }}
+          </p>
+          <p>{{ convertUnixTimeInMsToDateString(dataMetaInfo.uploadTime * 1000) }}</p>
+        </div>
+        <p class="mt-5">{{ dynamicButtonTitle }}</p>
+        <PrimeButton
+          class="uppercase p-button p-button-sm d-letters mt-3"
+          :disabled="!isFrontendUploadFormExisting"
+          label="Create Dataset"
+          icon="pi pi-plus"
+          @click="redirectToUploadForm"
+        />
+        <div v-if="!isFrontendUploadFormExisting">
+          <p>
+            (Uploading data for this framework is currently not enabled on the Dataland frontend. You can use the
+            Dataland API to do so.)
+          </p>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { convertUnixTimeInMsToDateString } from "@/utils/DateFormatUtils";
+import PrimeButton from "primevue/button";
+import { DataMetaInformation, DataTypeEnum } from "@clients/backend";
+
+export default defineComponent({
+  name: "MetaInfoPerComanyAndFramework",
+  components: { PrimeButton },
+
+  props: {
+    title: {
+      type: String,
+      required: true,
+    },
+    isFrontendViewPageExisting: {
+      type: Boolean,
+      default: true,
+    },
+    isFrontendUploadFormExisting: {
+      type: Boolean,
+      default: true,
+    },
+    frameworkUrlPath: {
+      type: String,
+      required: true,
+    },
+    isWaitingForData: {
+      type: Boolean,
+      default: true,
+    },
+    companyId: {
+      type: String,
+      required: true,
+    },
+    listOfFrameworkData: {
+      type: Array,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      convertUnixTimeInMsToDateString: convertUnixTimeInMsToDateString,
+    };
+  },
+
+  computed: {
+    dynamicDatasetTitle() {
+      if (this.isFrontendViewPageExisting) {
+        return this.title;
+      } else {
+        return `${this.title} (only viewable via API)`;
+      }
+    },
+    dynamicButtonTitle(): string {
+      if (this.listOfFrameworkData.length === 0) {
+        return "Be the first to create this dataset";
+      } else {
+        return `Create another dataset for ${this.title}`;
+      }
+    },
+  },
+  methods: {
+    /**
+     * Executes a router push to the page of the given data set if the framework it belongs to is enabled in the frontend
+     *
+     * @param dataMetaInfoOfClickedDataSet the meta information of the data set in question
+     */
+    async redirectToViewPageIfEnabledInFrontend(dataMetaInfoOfClickedDataSet: DataMetaInformation) {
```

Comment:
What does this do?

Reply:
In this Vue component, all existing datasets for a specific company and framework are displayed.
For example if BMW has one dataset for the framework "SME", you will see it here.
But if you click on it, you cannot be redirected to the view-page for that dataset, because the SME-view-page is not implemented yet in the frontend. SME is just available in the backend.

This method here first checks, if this Vue-component is mounted with the prop "isFrontendViewPageExisting" set to "true". Only then you shall be redirected to the view-page. Otherwise nothing happens.

Furthermore there is one check here, if you clicked on an lksg dataset. In this case you are redirected to the lksg view-page without any query param set. This makes sense because the lksg view page displays ALL datasets at once (unlike the eutaxonomy view-pages)

Reply:
Ok Thank you!